tests/playbooks/roles/install-csi-cinder/tasks/main.yaml

```diff
@@ -0,0 +1,213 @@
+---
+- name: Ensure CPO source folder
+  shell:
+    executable: /bin/bash
+    cmd: |
+      rm -rf $GOPATH/src/k8s.io/cloud-provider-openstack
+      mkdir -p $GOPATH/src/k8s.io; cd $_
+      git clone https://github.com/kubernetes/cloud-provider-openstack
+      cd cloud-provider-openstack
+      git fetch origin +refs/pull/{{ github_pr }}/merge
+      git checkout FETCH_HEAD; git checkout -b PR{{ github_pr }}
+
+- name: Build and upload cinder-csi-plugin image
+  shell:
+    executable: /bin/bash
+    cmd: |
+      cd $GOPATH/src/k8s.io/cloud-provider-openstack
+
+      REGISTRY={{ image_registry_host }} \
+      VERSION={{ github_pr }} \
+      IMAGE_NAMES=cinder-csi-plugin \
+      make upload-image-amd64
+
+- name: Prepare cloud config
+  shell:
+    executable: /bin/bash
+    cmd: |
+      kubectl -n kube-system get secret cloud-config >/dev/null 2>&1
+      if [ $? -eq 0 ]; then
+        exit 0
+      fi
+
+      set -ex
+
+      set +x; source {{ devstack_workdir }}/openrc admin admin > /dev/null; set -x
+      tenant_id=$(openstack project show demo -c id -f value)
+      sudnet_id=$(openstack subnet show private-subnet -c id -f value)
+      external_network_id=$(openstack network list --external -c ID -f value)
+
+      set +x; source {{ devstack_workdir }}/openrc demo demo > /dev/null; set -x
+      cat <<EOF > {{ ansible_user_dir }}/cloud.conf
+      [Global]
+      auth-url=${OS_AUTH_URL}/v3
+      username=${OS_USERNAME}
+      password=${OS_PASSWORD}
+      region=${OS_REGION_NAME}
+      tenant-id=$tenant_id
+      domain-id=default
+
+      [BlockStorage]
+      bs-version=v3
+      EOF
+
+      kubectl create secret -n kube-system generic cloud-config --from-file={{ ansible_user_dir }}/cloud.conf
+
+      # replace manifest cloud secret file
+      b64data=`cat {{ ansible_user_dir }}/cloud.conf | base64 -w 0`
+      cd $GOPATH/src/k8s.io/cloud-provider-openstack
+
+      sed -i "/cloud\.conf/c\  cloud.conf: $b64data" manifests/cinder-csi-plugin/csi-secret-cinderplugin.yaml
+
+      # replace image with built image
+      sed -i "s#docker.io/k8scloudprovider/cinder-csi-plugin:latest#{{ remote_registry_host }}/cinder-csi-plugin-amd64:{{ github_pr }}#" manifests/cinder-csi-plugin/cinder-csi-controllerplugin.yaml
+
+      sed -i "s#docker.io/k8scloudprovider/cinder-csi-plugin:latest#{{ remote_registry_host }}/cinder-csi-plugin-amd64:{{ github_pr }}#" manifests/cinder-csi-plugin/cinder-csi-nodeplugin.yaml
+
+- name: Deploy openstack-cloud-controller-manager
+  shell:
+    executable: /bin/bash
+    cmd: |
+      set -x
+
+      cat <<EOF | kubectl apply -f -
```

Comment:
why cant we directly use the one from the repo ? Is something changed in manifests?

Reply:
I compared ,  a few changes and I followed OCCM...

Reply:
I dont think we require those changes , as we are directly pulling from docker. Please check and update accordingly.